src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/ChangeTrackingService.kt

```diff
@@ -0,0 +1,168 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import jakarta.persistence.EntityNotFoundException
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.VideoBooking
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AmendVideoBookingRequest
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request.AppointmentType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.AdditionalBookingDetailRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.locations.LocationsService
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.UUID
+
+/**
+ * This service is used to identify and prevent unnecessary entity updates if there are no actual changes.
+ *
+ * It must be used before any changes are made to the existing entities.
+ */
+@Service
+class ChangeTrackingService(
+  private val videoBookingRepository: VideoBookingRepository,
+  private val locationsService: LocationsService,
```

Comment:
The location service would not be needed if the UI was passing location IDs.  Of course the API would also need to support location IDs.